src/core/toga/widgets/optioncontainer.py

```diff
@@ -3,6 +3,63 @@
 from .base import Widget
 
 
+class OptionItem:
+    def __init__(self, container, label, widget, enabled):
+        self._container = container
+        self._index = len(container)
+        self._label = label
+        self._widget = widget
+        self._enabled = enabled
+
+    @property
+    def enabled(self):
+        return self._enabled
+
+    @enabled.setter
+    def enabled(self, value):
+        self._enabled = value
+        self._container.interface._impl.set_option_enabled(self._index, value)
+
+    @property
+    def label(self):
+        return self._label
+
+    @label.setter
+    def label(self, value):
+        self._label = value
+        self._container.interface._impl.set_label(self._index, value)
+
+    def refresh(self):
+        self._widget.refresh()
+
+
+class OptionList:
+
+    def __init__(self, interface):
+        self.interface = interface
+        self._options = []
+
+    def __setitem__(self, index, option):
+        option.index = index
+        self._options[index] = option
+
+    def __getitem__(self, index):
+        return self._options[index]
+
+    def __delitem__(self, key):
+        del self._options[key]
+
+    def __iter__(self):
+        return iter(self._options)
+
+    def __len__(self):
+        return len(self._options)
+
+    def append(self, label, widget, enabled=True):
+        option = OptionItem(self, label, widget, enabled)
```

Comment:
Rather than (a) passing in self, and (b) having the constructor for OptionItem assume the position by evaluating len(), we should set `option.index` and `option.interface` explicitly before appending to `self._options`.